### DISCUSSION TITLE
On Site Fields, is there a UpdateAndPushChanges option?

### DISCUSSION THREAD

**Discussion by pmatthews05** [pnp/pnpcore#888] [Q&A] [answered]
I'm slowly trying to upgrade/re-writing an old SharePointPnPCoreOnline C# code to use the latest pnp.core.

Originally I had some code that updated the default value of the site column, this pushed the default value down to all the lists.

```csharp
Microsoft.SharePoint.Client.Field spfield = clientContext.Web.GetFieldByInternalName("myInternalName");
 spfield.DefaultValue = "myDefaultValue";
 spfield.UpdateAndPushChanges(true);
 await clientContext.ExecuteQueryRetryAsync();
```

In my latest code, I can update the default value, but it doesn't push changes.
```csharp
await pnpContext.Web.LoadAsync(p => p.Fields,
               p => p.ContentTypes.QueryProperties(p => p.Name,
               p => p.FieldLinks.QueryProperties(p => p.Name, p => p.FieldInternalName)));

var field = pnpContext.Web.Fields.AsRequested().FirstOrDefault(p => p.InternalName == "myInternalName");
field.DefaultValue = "myDefaultValue";
await field.UpdateAsync();
```

Is this a feature that doesn't exist, or am I missing a command?

**Comment by jansenbe** [comment, marked as answer]
@pmatthews05 : this is something we don't have today, but I'll add a work item to get this added. SharePoint REST is not offering an option for this but via [Graph](https://docs.microsoft.com/en-us/graph/api/columndefinition-update?view=graph-rest-1.0) or CSOM this is possible. Most likely will follow the Graph approach here.

**Reply by jansenbe** [reply]
@pmatthews05 : see https://github.com/pnp/pnpcore/issues/889

**Reply by jansenbe** [reply]
@pmatthews05 : this feature has been added and will be part of the next nightly and the upcoming 1.7 release (end of this week).

**Reply by pmatthews05** [reply]
This is excellent news, and that 1.7 will be out so soon. I'm also after the navigation on the IWeb.

**Reply by pmatthews05** [reply]
@jansenbe  Could you provide me a link to the code that you have implemented?

**Reply by jansenbe** [reply]
@pmatthews05 : here you go: https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Model/SharePoint/Core/Internal/Field.cs#L444-L525. Implementation is done by wrapping the CSOM method. For list fields (which could have been used in list content types) the regular update is used as there's no need to push changes down (it's done automatically)
